test/library/standard/Path/dlongnecke/normPath/normPath.chpl

```diff
@@ -0,0 +1,74 @@
+/*
+   Collapse paths such as `foo//bar`, `foo/bar/`, `foo/./bar`, and
+   `foo/baz/../bar` into `foo/bar`.  Warning: may alter meaning of paths
+   containing symbolic links.
+
+   .. note::
+
+      This is currently only implemented in a Unix environment.  It will not
+      behave correctly in a non-Unix environment.
+
+   :arg name: a potential path to collapse, possibly destroying the meaning of
+              the path if symbolic links were included.
+   :type name: `string`
+
+   :return: the collapsed version of `name`
+   :rtype: `string`
+*/
+
+
+use Path;
+
+
+// As per the examples given in the description of the normPath function.
+const collapsedA = 'foo/bar';
+const collapseToA0 = 'foo//bar';
+const collapseToA1 = 'foo/bar/';
+const collapseToA2 = 'foo/./bar';
+const collapseToA3 = 'foo/baz/../bar';
+
+// Variables for expanded coverage.
+const multiCurDir = '././.';
+const multiParentDir = '../../..';
+const multiMixedParentCurDir = './.././.././..';
+const multiOnlyPathSep = '/////';
+const pathSep2 = '//';
+
+
+proc collapse(name: string) {
+   writeln(name, ' becomes ', normPath(name));
+}
+
+
+// No change when given a valid path.
+collapse(collapsedA);
+// Collapse redundant separators.
+collapse(collapseToA0);
+// Eliminate trailing path separators.
+collapse(collapseToA1);
+// Eliminate current directory symbols.
+collapse(collapseToA2);
+// Properly collapse up-level references.
+collapse(collapseToA3);
+// Empty string yields single current dir.
+collapse('');
+// Single current dir yields single current dir.
+collapse(curDir);
+// Single parent dir yields single parent dir.
+collapse(parentDir);
+// Only current dirs yields single current dir.
```

Comment:
Minor nit, it might be good to have more line separation between each portion of the test.  I kinda lost which comment was associated with which code line at this point and had to look earlier in the file to remind myself.